File: graph/bfs.py
def bfs(u):
    visited[u] = True

    q = [u]

    while q != []:
        node = q.pop()
        print(node, end=' ')

        for neighbour in adj[node]:
            if not visited[neighbour]:
                visited[neighbour] = True
                q.append(neighbour)

v, e = list(map(int, input().split()))

adj = [[] for _ in range(v)]
visited = [False] * v

for i in range(e):
    a, b = list(map(int, input().split()))

    adj[a].append(b)
    adj[b].append(a)

bfs(0)

File: graph/bfs.cpp
#include <iostream>
#include <vector>
#include <queue>

#define MAX 100005

using namespace std;

vector<int> adj[MAX];
bool visited[MAX];

void addEdge(int a, int b)
{
    adj[a].push_back(b);
    adj[b].push_back(a);
}

void BFS(int u)
{
    visited[u] = true;

    queue<int> q;
    q.push(u);

    while (!q.empty())
    {
        int s = q.front();
        cout << s << ' ';
        q.pop();

        for (auto i: adj[s])
        {
            if (!visited[i])
            {
                visited[i] = true;
                q.push(i);
            }
        }
    }
}

int main()
{
    int v, e;

    cout << "Enter the number of vertices: ";
    cin >> v;
    cout << "Enter the number of edges: ";
    cin >> e;

    cout << "Enter the edge details:\n";

    int a, b;

    for (int i = 0; i < v; i++)
    {
        visited[i] = false;
        adj[i].clear();
    }

    while (e--)
    {
        cin >> a >> b;
        addEdge(a, b);
    }

    // Breadth-First-Search starting from vertex 0.
    BFS(0);

    return 0;
}